include/libcyphal/transport/can/media.hpp

```diff
@@ -110,6 +110,30 @@ class IMedia
     CETL_NODISCARD virtual PopResult::Type pop(const cetl::span<cetl::byte> payload_buffer) noexcept = 0;
     ///@}
 
+    /// @brief Registers "ready to push" callback function at a given executor.
```

Comment:
What does "ready to push" mean precisely. That is, ready to push what? An MTU-worth? one-byte?

Reply:
It means that IO output is ready to accept next frame for transmission. In case of posix implementation it means that socketcan file descriptor (passed to POSIX `::poll`) becomes "ready" (namely `POLLOUT` flag is set at `revents`). See also `ITxSocket::registerCallback` docs.

But if you have f.e. CAN media at micro controller then it probably means that some interrupt had happened (signaling that CAN hw tx queue is done transmitting previous frame, and "ready" to accept more).

Reply:
Can we update the verbiage where we discuss "ready to push" in the code to indicate this is about ready to push frames instead of just saying "data"?